README split, edits, examples (#346)
diffkemp/diffkemp: docs/examples/simple_program.md

```

## 3. Comparing the program using DiffKemp

Let's finally use DiffKemp to see how to use it to check if function behavior
changed. When we want to use DiffKemp we have to first create a *snapshot* of
each version of a program and then we can compare these snapshots.
The comparison will tell us which function semantics have probably changed.

> [!NOTE]
> Snapshot is the source file compiled to an intermediate code (LLVM IR) with
> additional metadata. When DiffKemp is comparing programs (checking its
> semantics), it compares the intermediate code to which the program was
> compiled.

### Creating a snapshot of the programs

To create the snapshot we will use DiffKemp
[`build` command](../usage.md#a-build-snapshot-generation-of-make-based-projects).
The command takes the path to the source file and the path to where it can
create the snapshot. We will build snapshots of both files and save them to
`old-snapshot` and `new-snapshot` directories.

```sh
diffkemp build src/old.c old-snapshot
diffkemp build src/new.c new-snapshot
```

### Comparing the program and interpreting the result

Now when we have created the snapshots of both programs, we can compare

> the programs (respectively the created snapshots). For this, we will use

Reviewer: ```suggestion
the programs (or more precisely, the created snapshots). For this, we will use
```

DiffKemp [`compare` command](../usage.md#2-semantic-comparison-command)
to which we will give paths to the snapshots. By default, the command will
compare all functions located in the file/program. We can also specify the
`--stdout` argument to get results to standard output.

```sh
diffkemp compare --stdout old-snapshot new-snapshot
```

Wait, there is no output?! That's right, similar to the classic diff utility,
DiffKemp `compare` command does not output anything if it evaluates the programs
(individual functions) as equal. Even if there is a syntactic difference
which we could see using the diff utility, the semantic of the functions
remained the same.

> ## 4. Making semantic difference

Reviewer: ```suggestion
## 4. Making a semantic difference
```

because in the function `add` was found semantic difference.

```txt
    Diff:
    *************** int add(int a, int b, int c) {
    *** 1,5 ***
      int add(int a, int b, int c) {
    !     int first = a;
    !     int second = b;
    !     return first + second;
      }
    --- 1,3 ---
      int add(int a, int b, int c) {
    !     return a + b + c;
      }
```

The last part of the output is again the syntax difference of the differing
(`add`) function.

```txt
    Callstack (old-snapshot):
    add at old.c:11

    Callstack (new-snapshot):
    add at new.c:9
```

We have skipped one part of the output which we did not see previously.
The part contains call stacks, which is a list of called functions from the
compared function (`mul`) to the differing function (`add`). For each function,
there is information about in which file was the function called and on which
line. We can see that the function `add` was called in the old version in the
`old.c` file on line `11` and in the new version in the file `new.c` on
line `9`. Since there are no other functions in the call stack, it indicates
that `add` was directly called from `mul`.

> To explain more the call stacks, consider a program structure like this:

Reviewer: ```suggestion
To explain the call stacks more, consider a program structure like this:
```
